tests/tracing/test_decorator_py3.py

```diff
@@ -101,3 +118,33 @@ async def test_trace_decorator_async_py3_no_trx():
                 "test_decorator_py3.my_async_example_function",
             )
             assert result2 == "return_of_async_function"
+
+
+def test_staticmethod_patching(sentry_init):
+    test_staticmethod_name = "test_decorator_py3.TestClass.static"
+    assert (
+        ".".join([TestClass.static.__module__, TestClass.static.__qualname__])
+        == test_staticmethod_name
+    ), "The test static method was moved or renamed. Please update the name accordingly"
+
+    sentry_init(functions_to_trace=[{"qualified_name": test_staticmethod_name}])
+
+    for instance_or_class in (TestClass, TestClass()):
+        with patch_start_child() as fake_start_child:
+            assert instance_or_class.static(1) == 1
+            fake_start_child.assert_called_once()
+
+
+def test_classmethod_patching(sentry_init):
+    test_classmethod_name = "test_decorator_py3.TestClass.class_"
+    assert (
+        ".".join([TestClass.class_.__module__, TestClass.class_.__qualname__])
+        == test_classmethod_name
+    ), "The test class method was moved or renamed. Please update the name accordingly"
+
+    sentry_init(functions_to_trace=[{"qualified_name": test_classmethod_name}])
+
+    for instance_or_class in (TestClass, TestClass()):
+        with patch_start_child() as fake_start_child:
+            assert instance_or_class.class_(1) == (TestClass, 1)
+            fake_start_child.assert_called_once()
```

Comment:
We should probably move these to `test_basics` since they are not specific to the test decorator and should also run on Python 2 (put them here originally when I thought I would fix the bug differently)

Reply:
Fine with me either way